src/org/folio/rest_v2/Consortia.groovy

```diff
@@ -100,12 +100,41 @@ class Consortia extends Authorization {
         OkapiTenantConsortia centralConsortiaTenant = consortiaTenants.find { it.isCentralConsortiaTenant }
         createConsortia(centralConsortiaTenant)
         addCentralConsortiaTenant(centralConsortiaTenant)
-        // Required sleep for 2 minutes because consorita async installation
-        sleep(1 * 60 * 1000)
+        checkConsortiaStatus(centralConsortiaTenant, centralConsortiaTenant)
         consortiaTenants.findAll { (!it.isCentralConsortiaTenant) }.each { institutionalTenant ->
             addConsortiaTenant(centralConsortiaTenant, institutionalTenant)
-            // Required sleep for 2 minutes because consorita async installation
-            sleep(1 * 60 * 1000)
+            checkConsortiaStatus(centralConsortiaTenant, institutionalTenant)
+        }
+    }
+
+    /**
+     * Check consortia mapped tenants status
+     *
+     * @param centralConsortiaTenant
+     *
+     * @param tenant
+     *
+     */
+    void checkConsortiaStatus(OkapiTenantConsortia centralConsortiaTenant, OkapiTenantConsortia tenant){
+        Map headers = getAuthorizedHeaders(centralConsortiaTenant)
+        String url = generateUrl("/consortia/${centralConsortiaTenant.consortiaUuid}/tenants/${tenant.tenantId}")
+        def response = restClient.get(url, headers, 5000).body
+        switch (response['setupStatus']){
+            case 'COMPLETED':
+                logger.info("Tenant : ${tenant.tenantId} added successfully")
+                break
+            case 'COMPLETED_WITH_ERRORS':
+                logger.warning("Tenant : ${tenant.tenantId} added with errors!")
+                break
+            case 'FAILED':
+                logger.error("Tenant : ${tenant.tenantId} add operation failed!\nAborting current execution!")
+                    steps.currentBuild.result = 'ABORTED'
```

Comment:
Move steps.currentBuild.result = 'ABORTED' before logger.error as this line will be never reached

Reply:
Hello Sir,
lines re-arranged thank you for your invaluable feedback!